src/main/scala/mesosphere/marathon/api/akkahttp/v2/EventsController.scala

```diff
@@ -1,42 +1,37 @@
 package mesosphere.marathon
 package api.akkahttp.v2
 
-import akka.NotUsed
-import akka.actor.Actor.Receive
-import akka.actor.{ ActorSystem, Cancellable }
+import akka.actor.ActorSystem
 import akka.event.EventStream
 import akka.http.scaladsl.model.RemoteAddress
 import akka.http.scaladsl.server.Route
-import akka.stream._
-import akka.stream.scaladsl.{ Sink, Source }
-import akka.stream.stage.GraphStageLogic.StageActor
-import akka.stream.stage.{ GraphStage, GraphStageLogic, OutHandler }
+import akka.stream.OverflowStrategy
+import akka.stream.scaladsl.Source
 import com.typesafe.scalalogging.StrictLogging
 import de.heikoseeberger.akkasse.ServerSentEvent
 import mesosphere.marathon.api.akkahttp.Controller
-import mesosphere.marathon.api.akkahttp.v2.EventsController.EventStreamSourceGraph
 import mesosphere.marathon.api.v2.json.Formats
+import mesosphere.marathon.core.async.ExecutionContexts
 import mesosphere.marathon.core.election.{ ElectionService, LeadershipState }
 import mesosphere.marathon.core.event.{ EventConf, EventStreamAttached, EventStreamDetached, MarathonEvent }
 import mesosphere.marathon.plugin.auth.{ Authenticator, AuthorizedResource, Authorizer, ViewResource }
+import mesosphere.marathon.stream.{ EnrichedFlow, EnrichedSource }
 import play.api.libs.json.Json
 
-import scala.collection.mutable
 import scala.concurrent.ExecutionContext
 import scala.concurrent.duration._
 
 /**
   * The EventsController provides a route to all MarathonEvents published via the event bus.
   */
 class EventsController(
-    val conf: EventConf,
-    val eventBus: EventStream)(
+    eventStreamMaxOutstandingMessages: Int,
+    eventBus: EventStream)(
     implicit
-    val actorSystem: ActorSystem,
     val executionContext: ExecutionContext,
     val authenticator: Authenticator,
     val authorizer: Authorizer,
-    val electionService: ElectionService
+    val electionService: ElectionService // TODO - it's inconsistent to make this an implicit
```

Comment:
Do we need it to be an implicit parameter?

Reply:
No :) It is just already that way for this and other controllers. I suggest we change in a separate commit.